Assets/BossRoom/Scripts/Shared/Game/Entity/NetworkCharacterState.cs

```diff
@@ -113,21 +113,25 @@ public LifeState LifeState
         public override void OnNetworkSpawn()
         {
             if (!IsServer) return;
-            HitPoints = GetInitialHitPoints();
+            SetInitialHitPoints();
         }
 
-        int GetInitialHitPoints()
+        void SetInitialHitPoints()
         {
+            HitPoints = CharacterClass.BaseHP.Value;
+
             if (!IsNpc)
             {
                 SessionPlayerData? sessionPlayerData = SessionManager<SessionPlayerData>.Instance.GetPlayerData(OwnerClientId);
                 if (sessionPlayerData is { HasCharacterSpawned: true })
                 {
-                    return sessionPlayerData.Value.CurrentHitPoints;
+                    HitPoints = sessionPlayerData.Value.CurrentHitPoints;
+                    if (HitPoints <= 0)
+                    {
+                        LifeState = IsNpc ? LifeState.Dead : LifeState.Fainted;
+                    }
```

Comment:
Feels to me like initializing `HitPoints` should be moved to `ServerCharacter` completely. You could just initialize the `LifeState` there on `OnNetworkSpawn()` based on `HitPoints` and isolate the setting of `LifeState` to just that class, since both should be server-driven.